More updates for the cal/val delivery

gshiroma opened the pull request:
This PR follows #35 and adds more updates on the RTC-S1 SAS for the cal/val delivery:
- Update RTC-S1 product metadata;
- Add option to dilate the layover/shadow mask;
- Add option to skip process if output geocoded files exist;
- Add layer name and description to GeoTIFF metadata;
- Update software version from gamma v0.3 to calval v0.4.

seongsujeong reviewed (commented):
Nice PR. My comments are as below - Seongsu

gshiroma commented on src/rtc/h5_prep.py:
```diff
@@ -269,12 +264,22 @@ def get_metadata_dict(product_id: str,
+              ' and the science content governed by the algorithm, input data,'
+               ' and processing parameter'],
+        'identification/productSpecificationVersion':
+            ['product_specification_version', product_version,
```
Thank you, @seongsujeong . We talked about it offline. I'm hard-coding the `productSpecificationVersion`.

gshiroma commented on src/rtc/rtc_s1_single_job.py:
```diff
@@ -891,7 +1057,40 @@ def run_single_job(cfg: RunConfig):
+            output_hdf5_file_burst = os.path.join(
+                hdf5_file_output_dir, f'{burst_product_id}.{hdf5_file_extension}')
+
+        # If burst imagery is not temporary, separate polarization channels
+        output_burst_imagery_list = []
+        if not flag_bursts_files_are_temporary:
+            for pol in pol_list:
+                geo_burst_pol_filename = \
+                    os.path.join(output_dir_bursts,
+                                 f'{burst_product_id}_{pol}.' +
+                                 f'{imagery_extension}')
+                output_burst_imagery_list.append(geo_burst_pol_filename)
+
+        else:
+            for pol in pol_list:
+                geo_burst_pol_filename = (f'NETCDF:{output_hdf5_file_burst}:'
+                                          '/data/'
+                                          f'{pol}')
+            output_burst_imagery_list.append(geo_burst_pol_filename)
```
HI @seongsujeong , yes, this is intentional. I'm separating the update of `output_burst_imagery_list` and the call to the function `_separate_pol_channels()` (which should happen after geocoding)

gshiroma commented:
> Nice PR. My comments are as below - Seongsu

Thank you, @seongsujeong . I addressed all your comments. Please, let me know if you have any other comments/concerns.

seongsujeong reviewed (commented):
Few more minor issues as below - S

seongsujeong commented on src/rtc/rtc_s1_single_job.py:
```diff
@@ -13,6 +13,7 @@
 import argparse
 
 import isce3
+from scipy import ndimage
```
Could not leave the comment for the line #5, but it seems the imported module `datetime` was not used in this code. Can you check?

seongsujeong commented on src/rtc/rtc_s1_single_job.py:
```diff
@@ -270,14 +320,45 @@ def append_metadata_to_geotiff_file(input_file, metadata_dict):
+    # Close GDAL dataset
     del gdal_ds
 
 
```
In the function `_separate_pol_channels()` below, please check if the parameter `pol_list`, and `output_radiometry_str` are used in the function.

gshiroma commented on src/rtc/h5_prep.py:
```diff
@@ -553,13 +571,35 @@ def get_metadata_dict(product_id: str,
+        ['source_data_zero_doppler_end_time',
+         burst_in.sensing_stop.strftime(DATE_TIME_FORMAT),
+         'Azimuth stop time of the product']  # 1.6.3
+    metadata_dict['metadata/sourceData/numberOfAzimuthLines'] = \
+        ['source_data_number_of_azimuth_lines',
+         burst_in.length,
+         'Number of azimuth lines within the source data product']
+    
+    metadata_dict['metadata/sourceData/numberOfRangeSamples'] = \
+        ['source_data_number_of_range_samples',
+         burst_in.width,
+         'Number of slant range samples for each azimuth line within the source data']
```
I think with the blank line it separates better the az and range metadata. Please, take a look at the file and let me know if you really prefer me to remove this line

seongsujeong replied:
I see. That makes sense. I just wanted to have the raster's width and length be separated by a line, but your way of code organization makes sense well. Please disregard my comment here.

gshiroma commented:
> Few more minor issues as below - S

Thank you, @seongsujeong . Just addressed them.
